voice-search-widget/InstantSearch.js/public/voice-widget/voice-widget.js

```diff
@@ -0,0 +1,131 @@
+import ChromeAPI from "./chrome-api.js";
+import GcpAPI from "./gcp-api.js";
+
+const chromeAPI = new ChromeAPI();
+const gcpAPI = new GcpAPI();
+
+class VoiceWidget {
+  constructor(options) {
+    Object.assign(this, options);
+    this.isTranscripting = {
+      state: false
+    }
+  }
+
+  init(initOptions) {
+    const voiceSearchContainer = document.querySelector(this.container);
+
+    voiceSearchContainer.innerHTML = `
+        <div id="voice-search-content" class="centered">
+          <div id="bars" class="hidden">
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+              <div class="bar"></div>
+          </div>
+          <div id="search-bar">
+            <input id="algolia-search-input" type="text" placeholder="${
+              this.placeholder
+            }" autocomplete="false" autofocus>
+          </div>
+          <button id="algolia-mic">
+            <i class="fas fa-microphone"></i>
+          </button>
+        </div>
+    `;
+
+    let mic = document.getElementById("algolia-mic");
+    let searchInput = document.getElementById("algolia-search-input");
+    let wave = document.getElementById("bars");
+
+    /*** Google Chrome API ***/
+    if (this._isChromeAPIAvailable()) {
+      chromeAPI.configureAPI(mic, searchInput, wave, initOptions, this.isTranscripting);
+    } else {
+      /*** GCP Speech-To-Text API ***/
+      gcpAPI.configureAPI(this.socket, searchInput, initOptions);
+    }
+
+    //Stopping the speech recognition if the user stopped talking
+    this.socket.on("endSpeechRecognition", _ => {
+      if (that.processor == "gcp") {
+        gcpAPI.stopTranscription(this.socket);
+      }
+      this.isTranscripting.state = false;
+      mic.innerHTML = '<i class="fas fa-microphone"></i>';
+      wave.classList.add("hidden");
+      searchInput.style.paddingLeft = "10px";
+    });
+
+    //Start/Stop mic on click
+    let that = this;
```

Comment:
You can [bind the function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) to avoid using this variable.